fms/utils/generation.py

```diff
@@ -144,22 +155,21 @@ def generate(
         num_beams: TODO: support beam search
         use_cache: requires that the model accept use_cache and
             past_key_value_states args in forward method.
+        eos_token_id: the optional token id representing the end of sequence
+        extra_kwargs: an optional mapping of additional kwargs to pass to the model
```

Comment:
might want to add detail on use-case like "if extra_kwargs contains `position_ids` and `mask` keys, these model parameters will be updated as-appropriate for each token generated"